### ISSUE TITLE
ARDUROVER

### ISSUE THREAD

**Issue by ghost** [MegaPirateNG/ardupilot-mpng#132]
Hello
will the APM:Rover 2.45  also ported to the crius v2 board ?

Regards MABA

**Comment by smurfy** [comment]
You can try to use the 3.0.1r4 branch and compile+flash the rover source. At least it should compile.
The thing is, the rover code is in between releases of the offical rover release.
If the code itself works on your vehicle, i can try to use one of the last stable versions of rover to compile for the board.

One issue could be the sensors, especially the sonar. if you use an offical supported from ardupilots rover, you should be fine. the cheap sonar is a problem. see: #32

**Comment by ghost** [comment]
thanks 
I will try it
MABA

**Comment by thomasbutler** [comment]
I compiled and flashed the 3.0.1r4 rover source on a Crius v1.1 board today. I only want to use it right now for manual control of a skid steer machine. It connected to MissionPlanner OK, and it sends telemetry via a USB cable; all sensors work (compass, gyro, accel, GPS, etc).  I'm not using sonar.  I set the parameters for skid steer. Unfortunately, the machine won't arm. I set all the six modes to "manual", so the machine should arm.  Oh well!

**Comment by thomasbutler** [comment]
Additionally, in the config.h sonar could come in on pin A0, batt V on A1, and batt I on A2

**Comment by thomasbutler** [comment]
and...in defines.h I set SONAR_SOURCE_ANALOG_PIN TO 0

**Comment by thomasbutler** [comment]
After these changes, connects to MissionPlanner, sets parameters, but still just boots and ends up with solid B (yellow on Crius board) LED on....Might be firmware thinks sonar has hit something; I'll try tying the sonar input (A0) low and then high.

**Comment by smurfy** [comment]
@thomasbutler don't have a vehicle yet (but just ordered) i will try to get the rover working too.
I'm not sure about the sonar problem. i only have a "hc-sr04" but i will try to get the sensor working too.

**Comment by ghost** [comment]
@smurfy Thank You for your support !!!

**Comment by thomasbutler** [comment]
Got the rover code loaded, but looks like a bug in the set_servo() function.  Sets motor1 at 4500\* and motor2 at 100 on line 190 in Settings.ino.  Output in skid steer had about 100ms wider pulse on Ch3. Ch1 and Ch3 not outputtting same values when throttle is increased running in "manual" mode. Cannot test now broken with code changed to both at 4500 because MissionPlanner won't connect to MPNG after compiling for rover and uploading...more to come.

**Comment by smurfy** [comment]
My car has arrived, but is a bit fast for at home :) I will compile a version and test it anyway soon.

**Comment by thomasbutler** [comment]
After much thrashing, still cannot get even PWM outputs in skid steer mode using 3.0.1r4.  However, if I set the PWM min, max, and trim for RC1 and RC3 I almost get equal output. However, max output at full throttle is only about 1650. Cannot get min values low enough on both channels.  Tested in both steering and manual mode.  I'm using the MissionPlanner to view the PWM values in the HUD.  At this point, I'm stuck.

**Comment by smurfy** [comment]
what car are you using?

**Comment by ghost** [comment]
Hello
Have you already ported the latest version ?
mababaar

**Comment by thomasbutler** [comment]
I finally got the rover code working in skid steer, but I've had to make some fixes.  It has one bug I've partially fixed; left motor runs up to 50% power when system is turned on and the Crius controller not yet fully initialized.   I've added some more nice features and fixes. One is an external LED connected to pin AN4.  "CopterLEDs" was not in the rover code. I'm in the process of porting the "copter LEDs" code and some of the features in ArduCopter to the rover code.  One especially important feature is the arming of the motors.  My base code is the MPNG v3.0.1r4.  I'm not going to v3.1 until is has had a long time to stabilize.  I have no plans to be the field tester of yet another version of half-baked ArduCopter. I had good luck with the code in my quadcopter using 3.0.1r4.  I'll post a set of the code which has the skid steer fixed ASAP.

**Comment by ghost** [comment]
@thomasbutler Thank you for this information
I hope your code will also work with my noname 4WD brushed monstertruck
I appreciate your work

Regards
mababaar

**Comment by ghost** [comment]
would sonar SR 04 be supported on this Version ?

**Comment by smurfy** [comment]
@mababaar nope, this is one of the reason i did not adapted the latest rover version, i'm restarted my #32 task getting cheap sonars working. (with help of some cheep extra hardware) attiny85 + 2 resistors probably.
